### ISSUE TITLE
Benchmark reference sizing results difference

### ISSUE THREAD

**Issue by Akilax0** [ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark#5]
Currently using the provided evaluation.py to test the given .size output files of the benchmark. However, I keep getting different values than the provided benchmark results at /design/README.md  

For example running 
 
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/46198141/eb0b0829-c9c3-4ad7-ba49-23fdcc39c13d)

would give the output as, 
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/46198141/244e8a28-5e88-4b85-ab91-7640b4968c91)

while the provided outputs are 
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/46198141/d1a13b1c-5ede-4088-8dff-61fc20a94ea7)

**Comment by bingyuew** [comment]
Please confirm you are using the updated benchmark.

**Comment by Akilax0** [comment]
Pulled again just now. Still the same results.

**Comment by bingyuew** [comment]
Please provide us with the benchmark you are using to reproduce the issue, including all necessary scripts.

**Comment by Akilax0** [comment]
I have direct cloned the repository.  https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/tree/main 

unzipped benchmark using the given script as [unzip.sh](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/blob/main/design/unzip.sh)

The script is unchanged : [evaluation.py](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/blob/main/src/evaluation/evaluation.py)

for file_path : ../../design/NV_NVDLA_partition_m/NV_NVDLA_partition_m.size
for design_name : NV_NVDLA_partition_m

(as in the first screenshot )

**Comment by bingyuew** [comment]
Please provide the files you are currently using so we can reproduce the issue.

**Comment by Akilax0** [comment]
[ICCAD_issue.zip](https://github.com/user-attachments/files/15845835/ICCAD_issue.zip)

README.md has the instructions to reproduce

**Comment by bingyuew** [comment]
I cannot reproduce your issue. Please rebuild the OpenROAD binary and confirm that you are using the most up-to-date version of OpenROAD.

**Comment by Akilax0** [comment]
I have rebuild and tested the same files on two separate machines. I keep getting the same output as above. 

The outputs you got for the provided files , are they the same as the given results at /design/README.md ?

**Comment by bingyuew** [comment]
Yes, I am getting the same result as in the README file. Please make sure you have updated the submodules.

**Comment by vickychen928** [comment]
@Akilax0 I encountered the same error as well. Have you managed to resolve it?
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/84194000/f46ab0a4-777d-4704-99f0-264804de59b0)

**Comment by bingyuew** [comment]
@vickychen928 @Akilax0  Could both of you share the message from your cmake and the process of building the openroad binary?

**Comment by vickychen928** [comment]
Here is the cmake and the process of building the openroad binary:
[cmake_log.txt](https://github.com/user-attachments/files/15897195/cmake_log.txt)
[make_log.txt](https://github.com/user-attachments/files/15897196/make_log.txt)

**Comment by Akilax0** [comment]
Yeah, I get the same values still. I also tried with the docker file same result. Here are my cmake and make outputs. 


[make_log.txt](https://github.com/user-attachments/files/15916008/make_log.txt)
[cmake_log.txt](https://github.com/user-attachments/files/15915998/cmake_log.txt)

**Comment by bingyuew** [comment]
Hi @Akilax0  and @vickychen928 , please try using this [Dockerfile.txt](https://github.com/user-attachments/files/15950696/Dockerfile.txt) and let me know if the problem is fixed.

**Comment by pipiofzl** [comment]
I encountered the same issue when running the provided example with OpenROAD_example.py. The output.size file generated was then used in the evaluation script evaluation.py, which produced the results. 
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/87072615/38c4c6ae-e047-4734-b7bd-ff5fde6ce76b)
Similarly, when I used the design/NV_NVDLA_partition_m/NV_NVDLA_partition_m.size file with the evaluation.py script, the results did not match the Statistics provided on the GitHub page for NV_NVDLA_partition_m.
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/87072615/ec21ab12-181b-445f-b089-249d4dc28429)
I would like to ask which programs were used to generate the Benchmark Statistics and the Reference Sizing Result Statistics provided on GitHub.

Thank you.

**Comment by bingyuew** [comment]
@pipiofzl Can you also share the message from your cmake and the process of building the openroad binary?

**Comment by vickychen928** [comment]
@bingyuew I would like to ask if I need to make changes because the version specified in docker.txt is Ubuntu 22.04, but the computing platform I am using is Ubuntu 20.04?

**Comment by pipiofzl** [comment]
Here is the cmake and the process of building the openroad binary:
cmake .. instruction:
[cmake.log](https://github.com/user-attachments/files/15966684/cmake.log)
make j 6 instruction:
[make_j_6.log](https://github.com/user-attachments/files/15966683/make_j_6.log)
make install instruction: 
[make_install.log](https://github.com/user-attachments/files/15966719/make_install.log)

Thank you.

**Comment by Akilax0** [comment]
@bingyuew  I got the same results as @pipiofzl  when using the Dockerfile 
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/46198141/91da6f71-e2f8-4a96-959d-c5c8b0c0d990)

**Comment by Akilax0** [comment]
@vickychen928  Shouldn't be an issue. Any reason why you asked this?

**Comment by vickychen928** [comment]
@Akilax0 There's no particular reason, just concerned whether specifying a different version from the platform might cause errors.

**Comment by bingyuew** [comment]
It will not cause an error as everyone is using the same base docker image

**Comment by bingyuew** [comment]
@Akilax0 @vickychen928 can you share the updated cmake log with the updated dockerfile from [here](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/issues/5#issuecomment-2185710800)?

**Comment by vickychen928** [comment]
@bingyuew I can't build the docker envirmnment. It stuck at installing dgl.
```bash
[+] Building 100.5s (29/44)                                                                                              docker:default
 => [internal] load .dockerignore                                                                                                  0.1s
 => => transferring context: 2B                                                                                                    0.0s
 => [internal] load build definition from Dockerfile                                                                               0.1s
 => => transferring dockerfile: 1.61kB                                                                                             0.0s
 => [internal] load metadata for docker.io/library/ubuntu:22.04                                                                    1.4s
 => [internal] load build context                                                                                                  0.1s
 => => transferring context: 32B                                                                                                   0.0s
 => [ 1/40] FROM docker.io/library/ubuntu:22.04@sha256:19478ce7fc2ffbce89df29fea5725a8d12e57de52eb9ea570890dc5852aac1ac            0.0s
 => CACHED [ 2/40] WORKDIR /src                                                                                                    0.0s
 => CACHED [ 3/40] RUN apt-get update && apt-get install -y gnupg2 ca-certificates                                                 0.0s
 => CACHED [ 4/40] RUN echo "deb [trusted=yes] https://downloads.skewed.de/apt jammy main" >> /etc/apt/sources.list                0.0s
 => CACHED [ 5/40] RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-key 612DEFB798507F25                                    0.0s
 => CACHED [ 6/40] RUN apt-get update                                                                                              0.0s
 => CACHED [ 7/40] RUN apt-get install -y git                                                                                      0.0s
 => CACHED [ 8/40] RUN apt-get install -y software-properties-common                                                               0.0s
 => CACHED [ 9/40] RUN add-apt-repository ppa:ubuntu-toolchain-r/test -y                                                           0.0s
 => CACHED [10/40] RUN apt-get update                                                                                              0.0s
 => CACHED [11/40] RUN apt-get install -y gcc-11 g++-11                                                                            0.0s
 => CACHED [12/40] RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 100                                          0.0s
 => CACHED [13/40] RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100                                          0.0s
 => CACHED [14/40] RUN apt-get install -y libpython-all-dev                                                                        0.0s
 => CACHED [15/40] RUN apt-get install -y libboost-all-dev                                                                         0.0s
 => CACHED [16/40] RUN apt-get install -y libcairo2                                                                                0.0s
 => CACHED [17/40] RUN apt-get install -y libcairo2-dev                                                                            0.0s
 => CACHED [18/40] RUN apt-get install -y python3-matplotlib                                                                       0.0s
 => CACHED [19/40] RUN apt-get install -y nvidia-cuda-toolkit                                                                      0.0s
 => CACHED [20/40] RUN apt-get update                                                                                              0.0s
 => CACHED [21/40] RUN apt-get install -y python3-graph-tool                                                                       0.0s
 => CACHED [22/40] RUN apt-get install -y vim                                                                                      0.0s
 => CACHED [23/40] RUN apt-get install -y python3-pip                                                                              0.0s
 => CACHED [24/40] RUN pip install --no-cache-dir torch==2.2.0                                                                     0.0s
 => ERROR [25/40] RUN pip install dgl==2.1.0                                                                                      98.8s
------
 > [25/40] RUN pip install dgl==2.1.0:
1.129 Collecting dgl==2.1.0
16.15   WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f0dbbc6b550>, 'Connection to files.pythonhosted.org timed out. (connect timeout=15)')': /packages/ff/45/44fea6e5228602f5d6ad8b4f5a377f5cf5cb5cb6444d9e7cd99150f13fa2/dgl-2.1.0-cp310-cp310-manylinux1_x86_64.whl
31.67   WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f0dbbc6b7f0>, 'Connection to files.pythonhosted.org timed out. (connect timeout=15)')': /packages/ff/45/44fea6e5228602f5d6ad8b4f5a377f5cf5cb5cb6444d9e7cd99150f13fa2/dgl-2.1.0-cp310-cp310-manylinux1_x86_64.whl
47.69   WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f0dbbc6b970>, 'Connection to files.pythonhosted.org timed out. (connect timeout=15)')': /packages/ff/45/44fea6e5228602f5d6ad8b4f5a377f5cf5cb5cb6444d9e7cd99150f13fa2/dgl-2.1.0-cp310-cp310-manylinux1_x86_64.whl
64.71   WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f0dbbc6baf0>, 'Connection to files.pythonhosted.org timed out. (connect timeout=15)')': /packages/ff/45/44fea6e5228602f5d6ad8b4f5a377f5cf5cb5cb6444d9e7cd99150f13fa2/dgl-2.1.0-cp310-cp310-manylinux1_x86_64.whl
83.73   WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f0dbbc6bc70>, 'Connection to files.pythonhosted.org timed out. (connect timeout=15)')': /packages/ff/45/44fea6e5228602f5d6ad8b4f5a377f5cf5cb5cb6444d9e7cd99150f13fa2/dgl-2.1.0-cp310-cp310-manylinux1_x86_64.whl
98.73 ERROR: Could not install packages due to an OSError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Max retries exceeded with url: /packages/ff/45/44fea6e5228602f5d6ad8b4f5a377f5cf5cb5cb6444d9e7cd99150f13fa2/dgl-2.1.0-cp310-cp310-manylinux1_x86_64.whl (Caused by ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f0dbbc6bdf0>, 'Connection to files.pythonhosted.org timed out. (connect timeout=15)'))
98.73 
------
Dockerfile:32
--------------------
  30 |     
  31 |     RUN pip install --no-cache-dir torch==2.2.0
  32 | >>> RUN pip install dgl==2.1.0
  33 |     RUN pip install pycairo
  34 |     RUN pip install pandas
--------------------
ERROR: failed to solve: process "/bin/sh -c pip install dgl==2.1.0" did not complete successfully: exit code: 1
```

**Comment by vickychen928** [comment]
@bingyuew I still got the wrong result.
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/84194000/6efd48ad-44ce-486a-a020-acaebc98fb57)

The following is my camke log:
[cmake_log_new.txt](https://github.com/user-attachments/files/16009457/cmake_log_new.txt)

**Comment by steveniscoming** [comment]
![image](https://github.com/ASU-VDA-Lab/2024_ICCAD_Contest_Gate_Sizing_Benchmark/assets/135108873/68038ff9-1078-46e8-9131-21d00b121c64)
me too 
get the same score for the reference partititon_m.size  and results did not match the Statistics provided on the GitHub page for NV_NVDLA_partition_m.

**Comment by Vicfretz** [comment]
was this issue resolved?

**Comment by lazzy-dog** [comment]
Is there really still no one who can fix this issue?

**Comment by steveniscoming** [comment]
I have try to download this repo on two server .One is equipped with gpu and the other isn't. The repo works correctly on the server with gpu but wrong on the one
without gpu. So I think it may have something to do with the gpu. Hope it would be helpful for you 🙂🙂

**Comment by lazzy-dog** [comment]
> I have try to download this repo on two server .One is equipped with gpu and the other isn't. The repo works correctly on the server with gpu but wrong on the one without gpu. So I think it may have something to do with the gpu. Hope it would be helpful for you 🙂🙂

Thanks for your help. Just to confirm: do I need to explicitly enable GPU support when running CMake, or is it sufficient that my PC has a GPU and the necessary drivers installed? My personal PC does have a GPU, and I noticed that target_compile_definitions(openroad PRIVATE GPU) is already present in the CMakeLists.txt.
Also—Long May the Sun Shine, my friend!

**Comment by steveniscoming** [comment]
> > I have try to download this repo on two server .One is equipped with gpu and the other isn't. The repo works correctly on the server with gpu but wrong on the one without gpu. So I think it may have something to do with the gpu. Hope it would be helpful for you 🙂🙂
> 
> Thanks for your help. Just to confirm: do I need to explicitly enable GPU support when running CMake, or is it sufficient that my PC has a GPU and the necessary drivers installed? My personal PC does have a GPU, and I noticed that target_compile_definitions(openroad PRIVATE GPU) is already present in the CMakeLists.txt. Also—Long May the Sun Shine, my friend!

I don't remember I had done any extra gpu setting  when I tried to running cmake. I just do as what Readme says.
 Did you still counter this problem even though  your personal PC does have a GPU ?

**Comment by lazzy-dog** [comment]
> I don't remember I had done any extra gpu setting when I tried to running cmake. I just do as what Readme says.
> Did you still counter this problem even though your personal PC does have a GPU ?

Yes, I just tried modifying my CUDA version and then recompiled, but the results didn’t seem to change. If it's convenient, could you please send me the version you are using to my email? I would really appreciate your help.
My email address is: zj.zhao.work@foxmail.com

**Comment by lazzy-dog** [comment]
> > > I have try to download this repo on two server .One is equipped with gpu and the other isn't. The repo works correctly on the server with gpu but wrong on the one without gpu. So I think it may have something to do with the gpu. Hope it would be helpful for you 🙂🙂
> > 
> > 
> > Thanks for your help. Just to confirm: do I need to explicitly enable GPU support when running CMake, or is it sufficient that my PC has a GPU and the necessary drivers installed? My personal PC does have a GPU, and I noticed that target_compile_definitions(openroad PRIVATE GPU) is already present in the CMakeLists.txt. Also—Long May the Sun Shine, my friend!
> 
> I don't remember I had done any extra gpu setting when I tried to running cmake. I just do as what Readme says. Did you still counter this problem even though your personal PC does have a GPU ?


I’ve made many attempts, but I still can’t reproduce the official results. It might be due to changes in OpenSTA affecting the calculations. I'm not sure how to proceed from here.😭😭
